services/provider/server/server.go

```diff
@@ -5,43 +5,202 @@ import (
 	"fmt"
 	"net"
 
+	ocsv1alpha1 "github.com/red-hat-storage/ocs-operator/api/v1alpha1"
+	"github.com/red-hat-storage/ocs-operator/services/provider/common"
 	pb "github.com/red-hat-storage/ocs-operator/services/provider/pb"
+	"github.com/red-hat-storage/ocs-operator/services/provider/server/consumer"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/reflection"
+	"google.golang.org/grpc/status"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/klog"
 )
 
-type ocsProviderServer struct {
+type OCSProviderServer struct {
```

Comment:
Why is this being done? I don't see it being used outside this package, so starting with lowercase would be more conventional.

Reply:
linter not happy if I do this. 
```
Running golangci-lint
services/provider/server/server.go:28:93: exported func NewOCSProviderServer returns unexported type *github.com/red-hat-storage/ocs-operator/services/provider/server.ocsProviderServer, which can be annoying to use (golint)
func NewOCSProviderServer(cClient consumer.APIUpdater, auth *AuthManager, namespace string) *ocsProviderServer {
```

Reply:
You cannot return a "private" (unexported type) from a "public" (exported) function
You either export both or none.

Reply:
marking as resolved.